solver/src/util.rs

```diff
@@ -20,3 +25,40 @@ macro_rules! chain {
         }
     };
 }
+
+pub fn bfp_to_big_decimal(num: Bfp) -> Result<BigDecimal> {
+    let rational = BigRational::from(num);
+    let top = BigDecimal::from_str(&rational.numer().to_string())?;
+    let bottom = BigDecimal::from_str(&rational.denom().to_string())?;
```

Comment:
I think there is a way to get rid of the string conversions and thus the Result. BigDecimal implements From<BigInt> and BigRational's numer and denom are BigInt so you can keep the division but go through BigInt.

```rust
let bottom = BigDecimal::from(rational.denom());
```

A potential problem is that BigDecimal is using an old version of the num crate (0.3 instead of 0.4) which makes the types incompatible.
The way to work around that is to depend on both versions of BigInt directly and convert the 0.4 BigInt into bytes with `to_bytes_le` and then create the 0.3 BigInt with `from_bytes_le`.

Reply:
I will try this out, but I believe I have already tried this and it was a conflicting BigInt crate.

Reply:
It appears to be a completely different crate `num_bigint` that implements `BigDecimal::from`

Reply:
Right, so what @vkgnosis was suggesting is that there are 2 `BigInt` types with the same name from different versions of the same crate. The compiler considers them different types internally (despite the same name, which can cause confusing compiler errors).

So, what you can do is:
```rust
use num::BigInt as BigInt04;
use num_bigint::BigInt as BigInt03;

let value_04: BigInt04 = // ...
let value_03 = BigInt03::from_bytes_le(value_04.to_bytes_le());
```

To convert between the types in an error-free way.

Reply:
Alrighty, we now have a new PR #813 which addresses this issue and will make the changes here sweet and smooth.